packages/jellyfish-api-core/__tests__/category/masternode/resignMasternode.test.ts

```diff
@@ -0,0 +1,101 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodeState } from '../../../src/category/masternode'
+import { RpcApiError } from '../../../src'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should resignMasternode', async () => {
+    const ownerAddress = await container.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(ownerAddress)
+
+    await container.generate(1)
+
+    const hex = await client.masternode.resignMasternode(masternodeId)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const resignedMasternode = Object.values(await client.masternode.listMasternodes()).filter(mn => mn.ownerAuthAddress === ownerAddress)
+
+    expect(resignedMasternode.length).toStrictEqual(1)
+    for (const masternode of resignedMasternode) {
+      expect(masternode.state).toStrictEqual(MasternodeState.PRE_RESIGNED)
+      expect(masternode.resignTx).toStrictEqual(hex)
+    }
+  })
+
+  it('should resignMasternode with utxos', async () => {
+    const ownerAddress = await container.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(ownerAddress)
+    const { txid } = await container.fundAddress(ownerAddress, 10)
+
+    await container.generate(1)
+
+    const utxos = (await container.call('listunspent'))
+      .filter((utxo: any) => utxo.txid === txid)
+      .map((utxo: any) => {
+        return {
+          txid: utxo.txid,
+          vout: utxo.vout
+        }
+      })
+
+    const hex = await client.masternode.resignMasternode(masternodeId, utxos)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const resignedMasternode = Object.values(await client.masternode.listMasternodes()).filter(mn => mn.ownerAuthAddress === ownerAddress)
+
+    expect(resignedMasternode.length).toStrictEqual(1)
+    for (const masternode of resignedMasternode) {
+      expect(masternode.state).toStrictEqual(MasternodeState.PRE_RESIGNED)
+      expect(masternode.resignTx).toStrictEqual(hex)
+    }
+  })
+
+  it('should resignMasternode with arbitrary utxos', async () => {
+    const ownerAddress = await container.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(ownerAddress)
+    const { txid, vout } = await container.fundAddress(ownerAddress, 10)
```

Comment:
Try this `const { txid, vout } = await container.fundAddress(await container.getNewAddress(), 10)`
   
 You should get 
 
```
 Error: RpcApiError: 'Test ResignMasternodeTx execution failed:
tx must have at least one input from the owner', code: -32600, method: resignmasternode
```

Reply:
Yes, i used the ownerAddress so naturally it will pass. Let me change that.